src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -182,8 +184,11 @@ export const createSavedQueryService = (
       },
     };
 
-    if (getUseNewSavedQueriesUI()) {
+    if (queryEnhancementEnabled) {
       savedQueryItem.query.dataset = savedQuery.attributes.query.dataset;
+    }
+
+    if (getUseNewSavedQueriesUI()) {
```

Comment:
queryEnhancementEnabled?

Reply:
No, for this we will use the new ui gate